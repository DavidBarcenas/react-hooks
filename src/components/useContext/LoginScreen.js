import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

  const {setUser} = useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr/>
      <button className="btn btn-success" 
        onClick={() => setUser({      
          id: 1234,
          name: 'Daveepro',
          email: 'daveepro@outook.com'})}
          >Login</button>
    </div>
  )
}
