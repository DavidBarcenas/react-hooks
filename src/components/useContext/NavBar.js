import React from 'react'
import { NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-light bg-dark">
        <div className="navbar-nav">
          
            <NavLink exact activeClassName="active" to="/">Home</NavLink>
            <NavLink exact activeClassName="active" to="/about">About</NavLink>
            <NavLink exact activeClassName="active" to="/login">Login</NavLink>
          
        </div>
    </nav>
  )
}
