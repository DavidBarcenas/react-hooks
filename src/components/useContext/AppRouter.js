import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { LoginScreen } from './LoginScreen'
import { HomeScreen } from './HomeScreen'

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path="/" component={HomeScreen} exact />
          <Route path="/about" component={AboutScreen} exact />
          <Route path="/login" component={LoginScreen} exact />
        </Switch>
      </div>
    </Router>
  )
}
