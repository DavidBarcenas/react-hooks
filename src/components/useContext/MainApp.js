import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

  const user = {
    id: 1234,
    name: 'Daveepro',
    email: 'daveepro@outook.com'
  }  

  return (
      <UserContext.Provider value={user}>
        <AppRouter />
      </UserContext.Provider>
  )
}
