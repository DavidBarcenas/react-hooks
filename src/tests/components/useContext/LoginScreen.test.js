import '@testing-library/jest-dom';
import React from 'react'
import { mount } from 'enzyme';
import { UserContext } from '../../../components/useContext/UserContext';
import { LoginScreen } from '../../../components/useContext/LoginScreen';

describe('<LoginScreen /> testing', () => {

  const setUser = jest.fn()

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  )

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  

  test('should run setUser with argument', () => {
    wrapper.find('button').simulate('click')
    expect(setUser).toHaveBeenCalledWith({
      id: 1234,
      name: 'Daveepro',
      email: 'daveepro@outook.com'
    })
  })
  
})
