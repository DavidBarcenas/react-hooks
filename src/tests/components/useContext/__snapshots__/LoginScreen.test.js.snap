// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> testing should render correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
